### PR TITLE
allow `and`, `or` and `with` operators

### PR DESCRIPTION
**Pull request by xsuchy**
This is follow up of discussion https://lists.spdx.org/g/spdx/message/1798 where this was found as good compromise https://lists.spdx.org/g/spdx/message/1817

And followup of https://github.com/spdx/spdx-spec/pull/876

### CODE REVIEW

File: docs/annexes/SPDX-license-expressions.md [hunk 1]
```diff
@@ -27,10 +27,16 @@ compound-expression = (simple-expression /
 
   simple-expression "WITH" addition-expression /
 
+  simple-expression "with" addition-expression /
```
Comment (zvr) [inline docs/annexes/SPDX-license-expressions.md:30]: please group the two versions in a single line:

```simple-expression ( "WITH" / "with" ) addition-expression```

File: docs/annexes/SPDX-license-expressions.md [hunk 2]
```diff
@@ -50,7 +56,7 @@ In the `tag:value` format, a license expression MUST be on a single line, and MU
 
 ## D.2 Case sensitivity <a name="D.2"></a>
 
-License expression operators (`AND`, `OR` and `WITH`) should be matched in a *case-sensitive* manner.
+License expression operators (`AND`, `and`, `OR`, `or`, `WITH` and `with`) should be matched in a *case-sensitive* manner. I.e., variations `And`, `oR` are not allowed.
```
Comment (zvr) [inline docs/annexes/SPDX-license-expressions.md:59]: I don't think the final explanation is needed.

If we want to have it in, reword it to something like:

> manner, i.e., letters must be all upper case or all lower case.

**Review by zvr — CHANGES_REQUESTED** [review]

**Comment by xsuchy** [comment]
Updated.

**Comment by xsuchy** [comment]
ping. @zvr can you please re-review?

**Review by zvr — APPROVED** [review]

**Review by Conan-Kudo — APPROVED** [review]

**Review by goneall — APPROVED** [review]
LGTM - Going ahead and merging
